src/storage/store/replication/post_commit_handler.rs

```diff
@@ -0,0 +1,59 @@
+use std::sync::Arc;
+
+use tokio::select;
+use tracing::error;
+
+use crate::storage::store::{engine::PostCommitMessage, replication::replicator::Replicator};
+
+struct PostCommitActor {
+    replicator: Arc<Replicator>,
+}
+
+impl PostCommitActor {
+    pub fn new(replicator: Arc<Replicator>) -> Self {
+        Self { replicator }
+    }
+
+    pub async fn run(&self, mut receive: tokio::sync::mpsc::Receiver<PostCommitMessage>) {
+        loop {
+            select! {
+                Some(msg) = receive.recv() => {
+                    if let Err(e) = self.replicator.handle_post_commit_message(&msg) {
+                        error!("Failed to handle post commit message: {}", e);
+                    }
+
+                    if let Err(e) = msg.channel.send(true) {
+                        error!("Failed to send post commit response: {}", e);
+                    }
+                }
+                else => break, // Exit loop if the channel is closed
+            }
+        }
+    }
+}
+
+pub struct PostCommitHandler {
+    pub sender: tokio::sync::mpsc::Sender<PostCommitMessage>,
+    handle: tokio::task::JoinHandle<()>,
+}
+
+impl PostCommitHandler {
```

Comment:
Is this used anywhere right now? I'm having a little trouble following where this fits in. I thought this recv would be happening closer to where the snapshot is taken.